pandas/tests/io/test_sql.py

```diff
@@ -476,17 +491,25 @@ def sqlite_conn(sqlite_engine):
 
 
 @pytest.fixture
-def sqlite_iris_str(sqlite_str, iris_path):
+def sqlite_iris_str(sqlite_str, iris_path, types_data):
```

Comment:
I think in the long run it would be better to not have separate `_iris` fixtures. The challenge is there is inconsistency across the drivers right now whether they work with the iris loading function (sqlite_building and sqlite_str being two that come to mind). So didn't try to change that much in this PR; leaving it to a future enhancement

Reply:
Hmm personally I would prefer to have separate fixtures for the connection and connection + iris

```python
@pytest.fixture
def connectable():
    with connection as con:
        yield conn
        conn.dump_all_tables_and_dispose()


@pytest.fixture
def connectable_with_iris(connectable):
      connectable.insert_iris_data()
      yield connectable
      connectable.delete_iris_data()
```

It makes it clear which tests specifically need iris data and ensures that there's no leftover state for subsequent tests if there's a failure

Reply:
Sounds good. No objection to that either - just will take a little more effort to get us there, which I don't plan on doing in this PR

Reply:
There's also the types table - do you think that should be a separate fixture or rolled into the iris one?

Reply:
> Sounds good. No objection to that either - just will take a little more effort to get us there, which I don't plan on doing in this PR

No worries, I can help out with this when I have spare cycles

> There's also the types table - do you think that should be a separate fixture or rolled into the iris one?

I don't remember how the types tables are used but my first reaction is that they should

Reply:
I think we can also make it so the iris / type table fixtures re-use the `all_connectables` fixture, instead of duplicating each connection for iris / type. So loosely would look something like this:

```python
@pytest.fixture
def postgres_conn():
    ...

@pytest.fixture
def sqlite_conn():
    ...

all_connectable = [postgres_conn, slqite_conn, ...]

@pytest.fixture
def iris_data(all_connectable):
    ...

@pytest.fixture
def types_data(all_connectable):
    ...
```

Today we have sqlite_conn_iris, postgres_conn_iris iterations and the types tables are loosely mixed in somewhere